components/compliance-service/reporting/relaxting/reports.go

```diff
@@ -846,10 +895,74 @@ func (backend *ES2Backend) getControlItem(controlBucket *elastic.AggregationBuck
 			controlSummary.Failed.Critical = total
 		}
 	}
+
+	if filteredWaivedStr, found := controlBucket.Aggregations.Filter("filtered_waived_str"); found {
+		if waivedStrBuckets, found := filteredWaivedStr.Aggregations.Terms("waived_str"); found && len(waivedStrBuckets.Buckets) > 0 {
+			contListItem.Waivers, err = backend.getWaiverData(waivedStrBuckets)
+			if err != nil {
+				return contListItem, err
+			}
+		}
+	}
+
 	contListItem.ControlSummary = controlSummary
 	return contListItem, nil
 }
 
+func (backend *ES2Backend) getWaiverData(waiverDataBuckets *elastic.AggregationBucketKeyItems) ([]*reportingapi.WaiverData, error) {
+	waiverDataCollection := make([]*reportingapi.WaiverData, 0)
+
+	for _, waiverDataBucket := range waiverDataBuckets.Buckets {
+		if aggResult, found := waiverDataBucket.Aggregations.Terms("expiration_date"); found && len(aggResult.Buckets) > 0 {
+			for _, expirationBucket := range aggResult.Buckets {
+				expDate, ok := expirationBucket.Key.(string)
+				if !ok {
+					logrus.Errorf("could not convert the value of expiration date: %v, to a string!", expirationBucket)
```

Comment:
What is the value of `expDate` if it fails to be converted to a string? Should we set `expDate` to ""?

Reply:
if ok is false then expDate will be the zero value of string which is ""